apps/trackers/tests/test_jira.py

```diff
@@ -1309,3 +1316,890 @@ def test_cve_severity_field(
             if flaw_impact == Impact.NOVALUE:
                 with pytest.raises(TrackerCreationError):
                     TrackerJiraQueryBuilder(tracker).generate()
+
+    @pytest.mark.parametrize(
+        "model_src,allowed_jira_src,expected_jira_src,other_outcome",
+        [
+            ("REDHAT", "Red Hat", "Red Hat", 0),
+            ("GIT", "Git", "Git", 0),
+            ("REDHAT", "RedHAT", "RedHAT", 0),  # Testing fallback allowed value pairing
+            ("REDHAT", None, None, 1),
+            ("REDHAT", "foobar", None, 2),
+        ],
+    )
+    def test_generate_source(
+        self, model_src, allowed_jira_src, expected_jira_src, other_outcome
+    ):
+        """
+        test that query has all fields correctly generated
+        """
+        JiraProjectFields.objects.filter(field_name="Source").delete()
+
+        JiraProjectFields(
+            project_key="FOOPROJECT",
+            field_id="customfield_12324746",
+            field_name="Source",
+            # Severely pruned for the test
+            allowed_values=["Foo", "Bar", allowed_jira_src, "Baz"],
+        ).save()
+
+        if other_outcome == 1:
+            JiraProjectFields.objects.filter(field_name="Source").delete()
+        if (
+            other_outcome == 1
+        ):  # TODO WHAT? probably delete this block, action 2 is probably wrong through input string
+            JiraProjectFields.objects.filter(field_name="Source").delete()
```

Comment:
I do not understand this but it seems I am not the only one :smile:

Reply:
Ooooh, sorry, I'll clean this up! An internal note, a reminder to double-check how to test the `other_outcome == 2` case, which I forgot to do and delete.